### PR TITLE
Meteor15 more tests

### PR DESCRIPTION
**Pull request by maritsandstad**
<Please replace this paragraph by a description of what this PR does and alter the TODO-list below to what applies here. Please make a 'draft PR' to begin and then only mark it as ready for review once all the relevant points are done.>

- [x] Tests added
- [x] Documentation added
- [x] Example added (in the documentation, to an existing notebook, or in a new notebook)
- [x] Description in ``CHANGELOG.rst`` added (single line such as: ``(`#XX <https://github.com/benmsanderson/METEOR/pull/XX>`_) Added feature which does something``)

### CODE REVIEW

File: src/meteor/pattern_logic_lib.py [hunk 1]
```diff
@@ -426,53 +428,6 @@ def wgt(array_w_lat):
     return weights
```
Comment (benmsanderson) [inline src/meteor/pattern_logic_lib.py:429]: Good call to remove this (was it ever used?)

File: src/meteor/meteor_interface.py [hunk 2]
```diff
@@ -1454,21 +1492,22 @@ def _apply_impacts(self, var_output, variable, impact_configs, custom_regions=No
                     elif key.startswith("regional:custom:"):
                         # Custom region - need to calculate baseline from bbox
                         from meteor import create_region_mask
```
Comment (benmsanderson) [inline src/meteor/meteor_interface.py:1494]: duplicated import?

File: src/meteor/meteor_interface.py [hunk 3]
```diff
@@ -1454,21 +1492,22 @@ def _apply_impacts(self, var_output, variable, impact_configs, custom_regions=No
...
                             mask = create_region_mask(picontrol_data, bbox=bbox)
                             masked_data = picontrol_data.where(mask)
                             baseline_k = float(masked_data.mean())
                         else:
-                            raise ValueError(f"Custom region '{region_name}' not found in custom_regions")
+                            raise ValueError(
+                                f"Custom region '{region_name}' not found in custom_regions"
+                            )
                     elif key.startswith("regional:"):
                         from meteor import regional_mean
```
Comment (benmsanderson) [inline src/meteor/meteor_interface.py:1507]: duplicated import

File: src/meteor/meteor_interface.py [hunk 4]
```diff
@@ -1454,21 +1492,22 @@ def _apply_impacts(self, var_output, variable, impact_configs, custom_regions=No
...
 
                         region = key.split(":")[1]
-                        baseline_k = float(
-                            regional_mean(picontrol_data, region).mean()
-                        )
+                        baseline_k = float(regional_mean(picontrol_data, region).mean())
                     elif key.startswith("point:"):
                         from meteor import extract_point
```
Comment (benmsanderson) [inline src/meteor/meteor_interface.py:1512]: and here, I think

File: src/meteor/geo_data_utils.py [hunk 5]
```diff
@@ -0,0 +1,491 @@
...
+        weights = weights / weights.mean()
+
+    # Calculate weighted mean
+    spatial_dims = [lat_name, lon_name]
+    weighted_data = dataset * weights
+    result = weighted_data.mean(spatial_dims, skipna=True)
+    return result
+
+
+def global_mean(
```
Comment (benmsanderson) [inline src/meteor/geo_data_utils.py:173]: This is much cleaner...

**Comment by benmsanderson** [comment]
Super nice. 

splitting prpatt.py is much better, makes things much more manageable. 
split geographic utilities from pattern scaling logic is perfect.

 weighting functions  are a lot more rational now (loss of wgt2() and wgt3(), simpler global_mean)

Tests:
 test_geo_data_utils.py - nice treatment of all varialbe names, edge cases, weight test is neat.
 test_pattern_logic_lib.py- efficient tests of functions/ parameter fitting/limts. 

Overall, significant increase in coverage - with logical tests, thank you :)).  

- minor comments only about duplicated imports, but maybe I'm missing somthing (and we might want to consider if the CDD code should be in a try statement - but this predates your PR).
- check errors seem to be in the notebooks, 

Further thoughts -beyond this PR:

- we could consider refactoring the split between meteor.py and meteor_interface.py.  I left latter as-is for backwards compatibility, but it's confusing for someone coming into it fresh.

But in general - this is great, fine to push the button from my end.

**Review by benmsanderson — APPROVED** [review]
See above comments.  Good to merge from my end. Thanks:))

File: src/meteor/meteor_interface.py (hunk 2 above)
Comment (maritsandstad) [reply src/meteor/meteor_interface.py:1494]: Yes, I haven't really gone over that file properly yet.

File: src/meteor/pattern_logic_lib.py (hunk 1 above)
Comment (maritsandstad) [reply src/meteor/pattern_logic_lib.py:429]: This is in use, not entirely sure I can remove it, the other version assumes a dataset, it might be an end goal down the line to move this out, but maybe not this minute...

File: src/meteor/meteor_interface.py (hunk 3 above)
Comment (maritsandstad) [reply src/meteor/meteor_interface.py:1507]: Yeah, I'll look at that file in the next PR

File: src/meteor/meteor_interface.py (hunk 4 above)
Comment (maritsandstad) [reply src/meteor/meteor_interface.py:1512]: Same, next PR...

File: src/meteor/geo_data_utils.py (hunk 5 above)
Comment (maritsandstad) [reply src/meteor/geo_data_utils.py:173]: Kind of the point of the exercise. This is how things should go though, first you write code that works, then you write code which is cleaner/easier to read/ more general / runs faster etc...

**Comment by maritsandstad** [comment]
I'll merge now. All comments are Welcome and useful, but I'll leave them for the next part to keep doing this in manageable chunks.
